src/Microsoft.AspNet.Mvc.Core/NonControllerAttribute.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// Indicates that the type and any derived types that this attribute is applied to
+    /// is not considered a controller by the default controller discovery mechanism.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = false, Inherited = true)]
```

Comment:
This is intentionally inherited. The suggestion would be to make `CustomController` abstract instead.